src/metadata/persistence/templates.clj

```diff
@@ -264,15 +329,9 @@
           (where {:id template-id}))
   nil)
 
-;; TODO: depending on how synonyms are going to be implemented, it may be necessary to check for existing synonyms
-;; before deleting the attributes as well.
-(defn- used-attribute-subselect
-  []
-  (subselect [:template_attrs :ta] (where {:ta.attribute_id :attributes.id})))
-
 (defn permanently-delete-template
   [template-id]
   (assert-found (get-metadata-template template-id) "metadata template" template-id)
   (delete :templates (where {:id template-id}))
-  (delete :attributes (where (not (exists (used-attribute-subselect)))))
+  (delete-orphan-attributes)
```

Comment:
I've seen some cases where deleting orphan attributes in this manner fails to catch all of the orphan attributes, thus leaving cruft behind in the database. This doesn't happen every time, so I believe it's a timing issue. Calling `delete-orphan-attributes` again does actually delete the orphaned attributes, so cruft will eventually be cleaned up. This behavior is suboptimal, but it shouldn't cause any problems. I'm going to leave it for the time being.